### ISSUE TITLE
The panoramic sky default field of view isn't set in orthogonal mode

### ISSUE THREAD

**Issue by capnm** [godotengine/godot#23077]
**Godot version:**

0dbe01483a902c49ecedf4fd36b74353424145a5

**OS/device including version:**

Linux

**Issue description:**

Auto/default FOV

![image](https://user-images.githubusercontent.com/4047289/47075911-d2e5b280-d1fd-11e8-8c11-ef9291928534.png)

![image](https://user-images.githubusercontent.com/4047289/47075951-edb82700-d1fd-11e8-9948-ac1ee65cad3d.png)

The same with a similar custom FOV

![image](https://user-images.githubusercontent.com/4047289/47076049-25bf6a00-d1fe-11e8-9378-60cdbb45c407.png)

![image](https://user-images.githubusercontent.com/4047289/47076084-38d23a00-d1fe-11e8-8714-309e890d4909.png)

**Steps to reproduce:**
Attach a panorama texture to the environment node and press numpad-5.

**Comment by NewNodeGames** [comment]
If I increase sky_custom_fov and then set back to zero the skybox or panorama looks correct. Weird.

**Comment by capnm** [comment]
![b1](https://user-images.githubusercontent.com/4047289/47090625-8dd47700-d223-11e8-9010-0b37e5d86fb3.png)

In Blender the background is at least not disturbing (and looks physically more correct ?):

![b2](https://user-images.githubusercontent.com/4047289/47090632-9331c180-d223-11e8-9160-3043cbae50e5.png)

**Comment by Calinou** [comment]
> The background flickers when you turn the view, but that doesn't seem to be a problem.

I tried that out as I was looking into fixing this, but I think the rapid color flashes when rotating can be problematic for people who are prone to light-induced seizures.

**Comment by Calinou** [comment]
Here's a patch that sets the sky FOV to 0.1 in orthogonal mode when no custom sky FOV is set:

```diff
diff --git a/servers/visual/rasterizer_rd/rasterizer_scene_high_end_rd.cpp b/servers/visual/rasterizer_rd/rasterizer_scene_high_end_rd.cpp
index a33c94fbcd..34a8ad2b57 100644
--- a/servers/visual/rasterizer_rd/rasterizer_scene_high_end_rd.cpp
+++ b/servers/visual/rasterizer_rd/rasterizer_scene_high_end_rd.cpp
@@ -1306,13 +1306,15 @@ void RasterizerSceneHighEndRD::_draw_sky(RD::DrawListID p_draw_list, RD::Framebu
 	// Camera
 	CameraMatrix camera;
 
-	if (custom_fov) {
+	if (custom_fov || p_projection.is_orthogonal()) {
+		// Use a very low FOV when using orthogonal projection.
+		// This is more technically correct compared to drawing a stretched out sky.
 
 		float near_plane = p_projection.get_z_near();
 		float far_plane = p_projection.get_z_far();
 		float aspect = p_projection.get_aspect();
 
-		camera.set_perspective(custom_fov, aspect, near_plane, far_plane);
+		camera.set_perspective(MAX(0.1, custom_fov), aspect, near_plane, far_plane);
 
 	} else {
 		camera = p_projection;
```

But as I said, it kind of hurts to look at when panning the camera around, especially in panoramas with strong light variations.

**Comment by capnm** [comment]
Now when I think about it again, in orthogonal mode it would be better to turn the panorama off for the background and/or make a viewport option for a replacement color. You can do that in the blender
viewport.

**Comment by DDL-Blue** [comment]
The sky in the very default "just opened Godot and created new project" 3D scene seems to have the same problem. When set to orthogonal view, the sun seems to be broken in the same way.

**Comment by Calinou** [comment]
@DDL-Blue Indeed, this issue isn't limited to PanoramaSky. It'll occur with any sky that's not a plain color.
